snippets/visualbasic/VS_Snippets_CLR_System/system.hashcode.structure/vb/example2.vb

```diff
@@ -0,0 +1,55 @@
+' <Snippet1>
+Public Structure Path
+    Implements IEquatable(Of Path)
+
+    Public ReadOnly Property Segments As IReadOnlyList(Of String)
+
+    Public Sub New(ParamArray ByVal segments() As String)
+        Me.Segments = segments
+    End Sub
+
+    Public Overrides Function Equals(obj As Object) As Boolean
+        Return (TypeOf obj Is Path) AndAlso Equals(DirectCast(obj, Path))
+    End Function
+
+    Public Overloads Function Equals(other As Path) As Boolean Implements IEquatable(Of Path).Equals
+        If ReferenceEquals(Segments, other.Segments) Then Return True
+        If Segments Is Nothing OrElse other.Segments Is Nothing Then Return False
+        If Segments.Count <> other.Segments.Count Then Return False
+
+        For i As Integer = 0 To Segments.Count - 1
+            If Not String.Equals(Segments(i), other.Segments(i)) Then Return False
+        Next
+
+        Return True
+    End Function
+
+    Public Overrides Function GetHashCode() As Integer
+        Dim hash As HashCode = New HashCode()
+
+        If Segments IsNot Nothing Then
```

Comment:
VB supports the `?.` operator for member access as well:

```visualbasic
For i as Integer = 0 To Segments?.Count - 1
    hash.Add(Segments(i))
Next
```

Several occurrences.